third_party/move/move-bytecode-verifier/src/signature_v2.rs

```diff
@@ -992,8 +993,11 @@ pub fn verify_module(module: &CompiledModule) -> VMResult<()> {
     res.map_err(|e| e.finish(Location::Module(module.self_id())))
 }
 
-pub fn verify_script(script: &CompiledScript) -> VMResult<()> {
-    let max_num = max_num_of_ty_params_or_args(BinaryIndexedView::Script(script));
+pub fn verify_script(config: &VerifierConfig, script: &CompiledScript) -> VMResult<()> {
+    let mut max_num = max_num_of_ty_params_or_args(BinaryIndexedView::Script(script));
+    if config.sig_checker_v2_fix_script_ty_param_count {
+        max_num = max_num.max(script.type_parameters.len());
+    }
```

Comment:
```suggestion
    let max_num = if config.sig_checker_v2_fix_script_ty_param_count {
        max_num = max_num.max(script.type_parameters.len())
    } else {
        max_num_of_ty_params_or_args(BinaryIndexedView::Script(script))
    }
```

Reply:
I kinda get what you're trying to change, but new logic you proposed doesn't seem quite right. Do you actually mean
```
let max_num = if config.sig_checker_v2_fix_script_ty_param_count {
        usize::max(max_num_of_ty_params_or_args(BinaryIndexedView::Script(script)), script.type_parameters.len())
    } else {
        max_num_of_ty_params_or_args(BinaryIndexedView::Script(script))

```
?